test.js

```diff
@@ -9,56 +9,64 @@ var url = 'mongodb://localhost:27017/mongocleantest'
 var baseCount = !process.env.MONGODB_VERSION || semver.satisfies(process.env.MONGODB_VERSION, '>= 3.2.0') ? 0 : 1
 
 function getDB (cb) {
-  MongoClient.connect(url, { w: 1 }, cb)
+  MongoClient.connect(url, { w: 1 }, (err, client) => {
+    if (err) return cb(err, null)
+    cb(null, client, client.db('mongocleantest'))
+  })
 }
 
-function close (db, t) {
-  db.close(function () {
+function close (client, t) {
+  client.close(function () {
     t.end()
   })
 }
 
-function cleanVerifyAndClose (db, t) {
-  clean(db, function (err, db) {
+function cleanVerifyAndClose (client, db, t) {
+  clean(db, function (err, cleanDb, cleanClient) {
```

Comment:
Can you also add promise support if a callback is not passed? I think it would be helpful with async/await.

Reply:
I thought about that, but since we are returning multiple values (the database and the client) we cannot support async await (not without returning an array and use destructuring on the other side).